xarray/coding/cftimeindex.py

```diff
@@ -149,10 +182,22 @@ class CFTimeIndex(pd.Index):
                                   'The microseconds of the datetime')
     date_type = property(get_date_type)
 
-    def __new__(cls, data):
```

Comment:
Let's keep the constructor the same here, rather than overriding it to match the signature of `cftime_range`. (I know this is a deviation from pandas, but frankly I think the pandas behavior is a confusing mistake.)

Reply:
Sure thing -- I retained my addition of the optional name argument so that we can support it in `cftime_range` (and obviously the constructor here as well).